common/persistence/elasticsearch/visibility_store.go

```diff
@@ -432,28 +436,59 @@ func (s *visibilityStore) ScanWorkflowExecutions(
 	}
 
 	ctx := context.Background()
-	var searchResult *elastic.SearchResult
-	var scrollService client.ScrollService
-	if len(token.ScrollID) == 0 { // first call
+	if clientV7, isV7 := s.esClient.(client.ClientV7); isV7 {
```

Comment:
maybe

```go
switch client := s.esClient.(type) {
case client.ClientV7:
    ...
case client.ClientV6:
    ...
default:
    ...
}

Reply:
There is no `ClientV6` interface, but I can use `default` for that. Let me try.

Reply:
Actually I can add `ClientV6` and move scroll to it.